tests/platform_tests/api/test_chassis.py

```diff
@@ -388,8 +388,9 @@ def test_sfps(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platf
         pytest_assert(sfp_list is not None, "Failed to retrieve SFPs")
         pytest_assert(isinstance(sfp_list, list) and len(sfp_list) == num_sfps, "SFPs appear to be incorrect")
 
-        for i in list_sfps:
-            sfp = chassis.get_sfp(platform_api_conn, i)
+        for i in range(num_sfps):
```

Comment:
@prgeor @lguohan 
we had extensive discussion on that with joe leveque , we found even though in link provided above the index should start from 0, we found lot of vendors had implementation that starts from index 1 along with our platform as it matches front panel port numbers. As per the exchange we had in emails we went and pushed changes to support both 0 and 1 based indexes the test case will read from paltform.json or port_config.ini whichever is present instead of assuming to be always 0.
Please refer to Joe's updated comment after discussing on the pr below

[https://github.com/Azure/sonic-mgmt/pull/2937](https://github.com/Azure/sonic-mgmt/pull/2937)

I am referring to his last comment especially:
`This is due to the fact that these vendors have defined their starting index to be 1 rather than 0 (example). If you look through all the platforms, you will see many which index their ports starting from 0. We suggest vendors index their ports starting from 0 (example). If you read the SONiC Porting Guide, "if the front panel ports are starting from 1, the index should start from 1. If the front panel ports start from 0, the index should start form 0. The break out ports from the same physical port should have same index.". Therefore, the base of this index is platform-specific.`

Reply:
I can't comment on the previous discussion. The common sfp test code should not worry about what is the mapping of physical port to index. The fact that the abstract class clearly indicates the SFP index should start from ZERO and someone not aware of the previous discussion would get confused. The reason is when the common code calls get_sfp(0) it means give me the FIRST SFP object that the platform thinks maps to FIRST physical port. get_sfp(1) should return the SFP object corresponding to "next" physical port and so on. 

           There is already a problem mentioned here #3886  if you start reading port_config.ini to do the mapping for physical port in the case of breakout. Though probably this can be fixed by adding a parser to distinguish between breakout port from physical port, I don't see why should the common test code be doing the mapping. The individual platform code should do the mapping as the test code is interested in only getting a list of 'N' SFP objects corresponding to 'N' physical port present in a platform. I would agree with @yxieca here to not merge [PR2937](https://github.com/Azure/sonic-mgmt/pull/2937). I would like to understand what is the challenge on the platform side to do device a mechanism that can do the mapping from zero based index to physical port number.